app.py

```diff
@@ -93,8 +97,7 @@ def get_user(id):
 
 @app.route('/users/<id>', methods=['DELETE'])
 def delete_user(id):
-    user = User.query.get_or_404(id)
-    user.deleted = True
+    User.query.get_or_404(id).soft_delete()
```

Comment:
You implemented the `soft_delete` method on the query object, but here you are calling it on a `User` instance. I don't understand how this is supposed to work.

Reply:
You are correct. I have changed it to work on the query. The only thing I'm not sure about is what we want to return from the soft_delete() function. For the moment I've changed it to return the list of modified objects, but we could also return a count of modified objects, or 'self' so that we don't guess what the user is trying to accomplish.

Reply:
I still fail to understand this change. Doesn't `get_or_404(id)` return an object of class `User`? You are calling `soft_delete()` on this object, not on a query. Or am I reading this incorrectly?

Reply:
You are reading correctly. On this version I had committed too quickly. I have since submitted a fix to this pull request that addresses the issue.

https://github.com/miguelgrinberg/sqlalchemy-soft-delete/pull/11/commits/51ccaa1c68047c17fdc98bbded664edc3c1ed85c